code/modules/mapfluff/ruins/spaceruin_code/hauntedtradingpost.dm

```diff
@@ -0,0 +1,362 @@
+
+/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield/Initialize(mapload)
+	. = ..()
+	GLOB.selfdestructs_when_boss_dies += src
+
+/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield/Destroy()
+	GLOB.selfdestructs_when_boss_dies -= src
+	return ..()
+
+//smes that produces power, until the boss dies then it self destructs and you gotta make your own power
+/obj/machinery/power/smes/magical/cybersun
+	name = "cybersun-brand power storage unit"
+	desc = "A high-capacity superconducting magnetic energy storage (SMES) unit. It looks like any other SMES unit, except this one says 'Cybersun' on it."
+	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
+	var/donk_ai_slave = FALSE
+
+/obj/machinery/power/smes/magical/cybersun/Initialize(mapload)
+	. = ..()
+	if(donk_ai_slave)
+		GLOB.selfdestructs_when_boss_dies += src
+
+/obj/machinery/power/smes/magical/cybersun/Destroy()
+	if(donk_ai_slave)
+		GLOB.selfdestructs_when_boss_dies -= src
+	return ..()
+
+//this is a trigger for traps involving doors and shutters
+//doors get closed and bolted, shutters get cycled open/closed
+/obj/machinery/button/door/invisible_tripwire
+	name = "Sonic Tripwire"
+	desc = "An invisible trigger for shutters and doors. Triggers when someone steps on the tile."
+	max_integrity = 50
+	invisibility = INVISIBILITY_ABSTRACT
+	anchored = TRUE
+	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
+	var/donk_ai_slave = FALSE
+	//can the trap trigger more than once?
+	var/multiuse = FALSE
+	//(if multiuse) how many times the trap can trigger. 0 or lower is infinite
+	var/uses_remaining = 0
+	//if true, the trap will unbolt all doors it bolted and cycle shutters a second time after a delay
+	var/resets_self = FALSE
+	//time before resets_self kicks in
+	var/reset_timer = 1.8 SECONDS
+	//when multiple tripwires are in the same suicide pact, they will all die when any of them die
+	var/suicide_pact = FALSE
+	//id of the suicide pact this tripwire is in
+	var/suicide_pact_id
+GLOBAL_LIST_EMPTY(tripwire_suicide_pact)
+
+/obj/machinery/button/door/invisible_tripwire/Initialize(mapload)
+	. = ..()
+	if(donk_ai_slave == TRUE)
+		GLOB.selfdestructs_when_boss_dies += src
+	if(suicide_pact == TRUE && suicide_pact_id != null)
+		GLOB.tripwire_suicide_pact += src
+	var/static/list/loc_connections = list(
+	COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/machinery/button/door/invisible_tripwire/proc/on_entered(atom/victim)
+	var/mob/living/target = victim
+	for(target in loc)
+		if(target.stat != DEAD && target.mob_size == MOB_SIZE_HUMAN && target.mob_biotypes != MOB_ROBOTIC)
+			tripwire_triggered(target)
+			if(multiuse && uses_remaining < 1)
+				uses_remaining--
+			if(resets_self)
+				addtimer(CALLBACK(src, PROC_REF(tripwire_triggered), victim), reset_timer)
```

Comment:
```suggestion
	SIGNAL_HANDLER
	if(!isliving(victim))
		return
	var/mob/living/target = victim
	if(target.stat != DEAD && target.mob_size == MOB_SIZE_HUMAN && target.mob_biotypes != MOB_ROBOTIC)
		tripwire_triggered(target)
		if(multiuse && uses_remaining < 1)
			uses_remaining--
		if(resets_self)
			addtimer(CALLBACK(src, PROC_REF(tripwire_triggered), victim), reset_timer)
```

Reply:
This makes it not work and I can't make it work again without reverting back to the other way I was doing it, so I'm going to keep it as is.

Reply:
It'd actually work, but it turn out you got the args of the comsig signal wrong, that's why!